include/rocksdb/options.h

```diff
@@ -2131,6 +2131,8 @@ struct CompactRangeOptions {
 // IngestExternalFileOptions is used by IngestExternalFile()
 struct IngestExternalFileOptions {
   // Can be set to true to move the files instead of copying them.
+  // Note that original file links will be removed after successful ingestion,
+  // unless `allow_db_generated_files` is true.
```

Comment:
Sorry I'm late looking at this but it seems very counter-intuitive. The simple contract of `move_files = true` based on that pre-existing sentence is that the ingest-from file entries will be gone if OK is returned. Why complicate a simple contract? This is especially weird if we foresee setting `allow_db_generated_files=true` by default in the future.

If you want the option to hard link files (without moving) rather than full copy them, then shouldn't that be an option like `link_when_available`? Is there a reason to limit that functionality to db generated files?

I don't think there's a serious danger of removing files from a live DB as long as `move_files = false` is the default, which should be forever because changing it would be a major semantic change. I think there's a greater risk of accidentally leaking externally written SST files that are combined with DB generated files, and neglecting that they weren't moved despite `move_files = true` (under the semantics in this PR).

Reply:
Thanks for the comment. I made this change when considering for files in a live DB, users might want to just hard link them without removing them. So the original DB can still function. But I agree that a dedicated option for this behavior is simpler. 

What do think about introducing a new option `link_files`, and allowing users to control whether to fall back to copy through option `failed_move_fall_back_to_copy`? `link_files` and `move_files` should not be set to true at the same time so `failed_move_fall_back_to_copy` should not be too ambiguous.

Reply:
I hadn't noticed `failed_move_fall_back_to_copy`, but yes I think that makes sense. The only thing I would add is to update the documentation of `move_files` to make it clear that it depends on hard links (LinkFile) to maximize the chances of being able to return to the original state on failure, rather than using a traditional move (RenameFile) which might be supported on some Filesystems that don't support links. In other words, the real difference between `move_files` and `link_files` is whether to unlink the source entries.